January 2024 releases blog post (#175)
conda-incubator/conda-dot-org: blog/2024-01-31-january-releases.mdx

---
title: "January 2024 Releases"
slug: "2024-01-31-january-releases"
authors: [kenodegard]
tags: [announcement, conda, conda-build, conda-libmamba-solver, conda-index]
description: |
  Conda 24.1.0/.1/.2, conda-build 24.1.0/.1/.2, conda-libmamba-solver 24.1.0, and conda-index 0.4.0 have been released! 🎉
image: img/blog/2024-01-31-january-releases/banner.jpg
---

The January 2024 releases included updates to four conda community projects: conda, conda-build, conda-libmamba-solver, and conda-index! 🎉 Each of these have been released to both `main` and `conda-forge`.

## Changes in Conda [24.1.0](https://github.com/conda/conda/releases/tag/24.1.0)/[24.1.1](https://github.com/conda/conda/releases/tag/24.1.1)/[24.1.2](https://github.com/conda/conda/releases/tag/24.1.2)

To update conda, run:

```bash
conda install -n base conda=24.1.2
```

### ✨ What's New? ✨

> - Verify signatures on to-be-installed packages instead of on all packages.
> - Add new `pre-solves` and `post-solves` plugin hooks.
> - Add support for Python 3.12.

Reviewer: ```suggestion
- Signature verification on to-be-installed packages instead of on all packages.
- New `pre-solves` and `post-solves` plugin hooks.
- Python 3.12 support.
```

- Check `repodata.json.zst` for faster repodata downloads.

> - Add `--skip-flexible-search` option in `conda search` to skip flexible search.

Reviewer: ```suggestion
- A new `--skip-flexible-search` option in `conda search` to skip flexible search.
```

- Provide a more useful warning when attempting to rename a non-existent prefix.
- Add a new flag `--keep-env` to be used with `conda remove --all`. It allows users to delete all packages in the environment while retaining the environment itself.
- Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but the entirety of the specified environment.
- Add `--repodata-use-zst/--no-repodata-use-zst` flag to control `repodata.json.zst` check; corresponding `repodata_use_zst: true/false` for `.condarc`. Default is to check for `repodata.json.zst`. Disable if remote returns unparsable `repodata.json.zst` instead of correct data or 404.

### 🔧 What Got Fixed? 🔧

- Create the `~/.conda` directory before trying to write to the `environments.txt` file.
- Ensure `PackageRecord.timestamp` is dumped in milliseconds.
- Fix error when setting a non-default repodata filename via `CONDA_REPODATA_FNS`.
- Fix the config file location where the integrated Anaconda client gateway loads user configuration from. This is a regression introduced in conda 23.11.0 when the `platformdirs` library was adopted.

> - Interpret missing `Cache-Control` header as `max-age=0` instead of exception.

Author: If it's unspecified it's the initial version, it was just too noisy to include `(24.1.0)` for all of the initial changes

code into the `conda` package, while allowing old `conda env` commands to still work via Python
import redirects. This is a first step of many to improving the user experience of the conda
command line interface related to environment management.

### 📄 What's New in Documentation? 📄

- Update [Getting Started](https://docs.conda.io/projects/conda/en/stable/user-guide/getting-started.html) documentation in User Guide.
- Add [GoatCounter](https://docs-conda-io.goatcounter.com/) for docs site analytics.
- Improve type hints and doc strings.

---

## Changes in Conda-Build [24.1.0](https://github.com/conda/conda-build/releases/tag/24.1.0)/[24.1.1](https://github.com/conda/conda-build/releases/tag/24.1.1)/[24.1.2](https://github.com/conda/conda-build/releases/tag/24.1.2)

To update conda-build, run:

```bash
conda install -n base conda-build=24.1.2
```

### ✨ What's New? ✨

- Update `conda inspect channels` to use updated solver/transaction logic.
- Relax `script_env` error in outputs when variable referenced in `script_env` is not defined. This unifies current behavior with the top-level build.

> - Add support for Python 3.12.

Reviewer: ```suggestion
- Python 3.12 support.
```

> - Adopt calendar versioning (CalVer) per CEP-8 for consistency with conda.

Reviewer: ```suggestion
- Adopt calendar versioning (CalVer) per [CEP-8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md) for consistency with conda.
```

> - Adopt expedited CEP-9 deprecation policy.

Reviewer: ```suggestion
- Adopt expedited [CEP-9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md) deprecation policy.
```

### 🔧 What Got Fixed? 🔧

- (24.1.1) Fix nonzero exitcode on success.

> - (24.1.2) Fix rpaths patcher being run on symbolic links.

Reviewer: ```suggestion
- (24.1.2) Fix `rpaths` patcher being run on symbolic links.
```
